### ISSUE TITLE
Prefer desktop instagram images if resolution is the same

### ISSUE THREAD

**Issue by qsniyg** [qsniyg/maxurl#360]
According to Regis on discord, the quality is usually (always?) higher

**Comment by schleeb** [comment]
Will this also allow access to the 1440 images (when available) over the 1080? See https://github.com/instaloader/instaloader/issues/630 for information if what I'm talking about isn't clear.

**Comment by qsniyg** [comment]
@schleeb For that it already does provide 1440! :) (if you're logged in, that is)

This issue is just tracking video thumbnails, as for some reason the website version gives higher quality thumbnails than the app version (even if the app's video is higher quality than the website's video)

**Comment by schleeb** [comment]
Really @qsniyg? I've never had it work (but I've confirmed 1440 images were available via instaloader); I've only ever been able to get the 1080 images. Is there a trick? Am I doing something wrong? lol

**Comment by qsniyg** [comment]
@schleeb Weird... are you logged into Instagram?

Can you try going to: https://www.instagram.com/p/CD9EMe5sHP5/, opening the popup (e.g. shift+alt+i, or whatever the trigger is set to)? It should return a 1349x1687 image (which is the largest size available for that specific image, though other images are 1440x1800)

**Comment by schleeb** [comment]
Nope; I even disabled any other possible conflicting TamperMonkey scripts, Brave's tracker blocking, and disabled ScriptSafe (I did this after I took the screenshot).
![Test Image](https://i.imgur.com/NbURkud.jpg "Test Image")

**Comment by qsniyg** [comment]
@schleeb Wow that's strange... could you press F12 to open the console, reload the page, try again, and see if there are any errors that are printed?

By the way, if you have a discord account, it might be a little easier to do the back-and-forth over there (https://discord.gg/fH9Pf54), as I think this issue might require a fair bit of it haha... :)

**Comment by schleeb** [comment]
Good call; I'm getting:
![Dev Console Errors](https://i.imgur.com/8Lb8K8v.jpg "Dev Console Errors")

I was getting the "429 Too Many Requests" issue w/ Instaloader as well (At least, I assume that's what the status: 429 is in reference to). I even tried flushing out any logged in instance, so currently it's just this browser...

I might try again in a few hours. I've got some stuff I need to take care of, but I appreciate the fast support! I should have a discord account setup somewhere.

**Comment by qsniyg** [comment]
@schleeb Interesting... yeah I'd say wait a few hours, and if there's anything else that's using the account (e.g. scrapers), make sure they're not running for a while, because their rate limiting is adaptive/ML-based to detect bots.

**Comment by schleeb** [comment]
Yeah; nothing else should be making requests. It's weird, but we'll see what happens in a bit.

**Comment by schleeb** [comment]
It's been 5 hours, but I'm still getting the 429. As I'd mentioned, I was getting this issue w/ Instaloader as well (so it's not specific to you). There's probably a way to grab this info from the dev console for MaxURL, but I don't know where, so I'll share what I got from Instaloader when I tried it recently:
```
GraphQL requests:
  d6f4427fbe92d846298cf93df0b937d3
    last 10 minutes: 1 requests
    last 11 minutes: 1 requests
    last 15 minutes: 1 requests
    last 20 minutes: 1 requests
    last 30 minutes: 1 requests
    last 60 minutes: 1 requests
  other
    last 10 minutes: 1 requests
    last 11 minutes: 1 requests
    last 15 minutes: 1 requests
    last 20 minutes: 1 requests
    last 30 minutes: 1 requests
    last 60 minutes: 1 requests
  iphone
    last 10 minutes: 2 requests
    last 11 minutes: 2 requests
    last 15 minutes: 2 requests
    last 20 minutes: 2 requests
    last 30 minutes: 2 requests
    last 60 minutes: 2 requests
  e31a871f7301132ceaab56507a66bbb7
    last 10 minutes: 1 requests
    last 11 minutes: 1 requests
    last 15 minutes: 1 requests
    last 20 minutes: 1 requests
    last 30 minutes: 1 requests
    last 60 minutes: 1 requests
  2b0673e0dc4580674a88d426fe00ea90
    last 10 minutes: 1 requests
    last 11 minutes: 1 requests
    last 15 minutes: 1 requests
    last 20 minutes: 1 requests
    last 30 minutes: 1 requests
    last 60 minutes: 1 requests
```

I could be misinterpreting what that's referring to, but I assume it's a reference to requests from my account and the session/source, but at the time I made that request, I'd only used the browser, and then Instaloader. I don't even have an iPhone, and my account only shows the two sessions logged in.

If I have time tomorrow, I'll try to hit you up on discord.

**Comment by schleeb** [comment]
After waiting through the night, it's working! I still wish I knew more about what triggered the original issue, and I think I was originally using the plugin wrong in the first place, but now I know, and I greatly appreciate your incredibly prompt and effective help!

**Comment by qsniyg** [comment]
@schleeb Glad to know it's fixed! Yeah Instagram can be _really_ weird at times haha. There's a reason it's the largest rule in the script (~1.3k loc), and _still_ needs constant maintenance in order to properly work hahaha

**Comment by fosam** [comment]
I have the same issue, the largest images only render to 1080x1080 or 1080x1351
When i side-loaded the extension i got the highest resolution, but after some hours of use they lowered to 1080x1080
Do i need to wait or am i doing something wrong? i reverted all the settings to their defaults and still not working

**Comment by qsniyg** [comment]
>When i side-loaded the extension i got the highest resolution, but after some hours of use they lowered to 1080x1080

This likely means you've been rate limited (so wait a few hours before trying again). Be careful with how many images you request with the app API, otherwise IG will rate limit you and possibly even flag your account.

Just to be sure though, can you check the console (f12)? In theory, it should show an error. Check if you can see a status code of 429 somewhere.
